Essentials/src/main/java/com/earth2me/essentials/signs/EssentialsSign.java

```diff
@@ -154,6 +156,35 @@ public String getUsername(final User user) {
         return user.getName().substring(0, Math.min(user.getName().length(), 13));
     }
 
+    public void setOwner(final IEssentials ess, final User user, final ISign signProvider, final int nameIndex, final String namePrefix) {
+        setOwnerData(ess, user, signProvider);
+        signProvider.setLine(nameIndex, namePrefix + getUsername(user));
+    }
+
+    public void setOwnerData(final IEssentials ess, final User user, final ISign signProvider) {
+        if (ess.getSignDataProvider() == null) {
+            return;
+        }
+        final Sign sign = (Sign) signProvider.getBlock().getState();
+        ess.getSignDataProvider().setSignData(sign, "owner", user.getUUID().toString());
+    }
+
+    public boolean isOwner(final IEssentials ess, final User user, final ISign signProvider, final int nameIndex, final String namePrefix) {
+        final Sign sign = (Sign) signProvider.getBlock().getState();
+        if (ess.getSignDataProvider() == null || ess.getSignDataProvider().getSignData(sign, "owner") == null) {
+            if (ess.getSignDataProvider() != null) {
+                ess.getSignDataProvider().setSignData(sign, "owner", user.getUUID().toString());
```

Comment:
This check changes the owner to the user without verifying who the user is first.